src/EditorFeatures/Core.Wpf/Completion/Presentation/RoslynCompletionSet.cs

```diff
@@ -215,16 +229,19 @@ public override IReadOnlyList<Span> GetHighlightedSpansInDisplayText(string disp
                 return null;
             }
 
+            var textForBolding = _displayTextToBoldingTextMap.TryGetValue(displayText, out var matchingText) ? matchingText : displayText;
+            Debug.Assert(displayText.Contains(textForBolding));
+
             var pattern = this.FilterText;
             if (_highlightMatchingPortions && !string.IsNullOrWhiteSpace(pattern))
             {
                 var completionHelper = this.GetCompletionHelper();
                 if (completionHelper != null)
                 {
                     var highlightedSpans = completionHelper.GetHighlightedSpans(
-                        displayText, pattern, CultureInfo.CurrentCulture);
+                        textForBolding, pattern, CultureInfo.CurrentCulture);
 
-                    return highlightedSpans.SelectAsArray(s => s.ToSpan());
+                    return highlightedSpans.SelectAsArray(s => new Span(s.Start + Math.Max(0, displayText.IndexOf(textForBolding)), s.Length));
```

Comment:
this is not good:

1. you are calling "displayText.IndexOf(textForBolding)" multiple times.  and this is in completoin where perf matters and the bolding codepath impacted perf negatively for quite a while.
2. this assumes that textForBolding is contained in displayText.  If not, this will just produce broken highlights.  

If you want to go this path (which i would not recommend), you should *first* ensure that textForBolding is contained within displayText.  If so, you can take this approach.  Otherwise you should highlight hte actual display text.